tests/test_hf_api.py

```diff
@@ -212,26 +212,26 @@ def test_delete_repo_missing_ok(self) -> None:
 
     def test_create_update_and_delete_repo(self):
         repo_id = self._api.create_repo(repo_id=repo_name()).repo_id
-        res = self._api.update_repo_visibility(repo_id=repo_id, private=True)
-        assert res["private"]
-        res = self._api.update_repo_visibility(repo_id=repo_id, private=False)
-        assert not res["private"]
+        self._api.update_repo_settings(repo_id=repo_id, private=True)
```

Comment:
For tests that are using `update_repo_visibility`, it's best to keep them as they are now and use the `@expect_deprecation` decorator (see comment above). The philosophy behind it is that we shouldn't change existing behavior for end users so we shouldn't change the tests. 

It's only when we will definitely remove the method from the library that we will modify the tests.